src/Parsers/Lexer.cpp

```diff
@@ -44,6 +45,35 @@ Token quotedString(const char *& pos, const char * const token_begin, const char
     }
 }
 
+Token quotedHexOrBinString(const char *& pos, const char * const token_begin, const char * const end)
+{
+    constexpr char quote = '\'';
+
+    assert(pos[1] == quote);
+
+    bool hex = (*pos == 'x' || *pos == 'X');
+
+    pos += 2;
+
+    if (hex)
+    {
+        while (pos < end && isHexDigit(*pos))
+            ++pos;
+    }
+    else
+    {
+        pos = find_first_not_symbols<'0', '1'>(pos, end);
+    }
+
+    if (pos >= end || *pos != quote)
+    {
+        return Token(TokenType::ErrorSingleQuoteIsNotClosed, token_begin, end);
```

Comment:
The problem is here, the `pos` should be set to the `end`, otherwise it will try to process tokes after.

```suggestion
        pos = end;
        return Token(TokenType::ErrorSingleQuoteIsNotClosed, token_begin, end);
```